Userland/Libraries/LibC/wctype.h

```diff
@@ -6,12 +6,153 @@
+static inline int __inline_iswalnum(wint_t wc)
+{
+    return __inline_isalnum(wc);
+}
+
+static inline int __inline_iswalpha(wint_t wc)
+{
+    return __inline_isalpha(wc);
+}
+
+static inline int __inline_iswcntrl(wint_t wc)
+{
+    return __inline_iscntrl(wc);
+}
+
+static inline int __inline_iswdigit(wint_t wc)
+{
+    return __inline_isdigit(wc);
+}
+
+static inline int __inline_iswxdigit(wint_t wc)
+{
+    return __inline_isxdigit(wc);
+}
+
+static inline int __inline_iswspace(wint_t wc)
+{
+    return __inline_isspace(wc);
+}
+
+static inline int __inline_iswpunct(wint_t wc)
+{
+    return __inline_ispunct(wc);
+}
+
+static inline int __inline_iswprint(wint_t wc)
+{
+    return __inline_isprint(wc);
+}
+
+static inline int __inline_iswgraph(wint_t wc)
+{
+    return __inline_isgraph(wc);
+}
+
+static inline int __inline_iswlower(wint_t wc)
+{
+    return __inline_islower(wc);
+}
+
+static inline int __inline_iswupper(wint_t wc)
+{
+    return __inline_isupper(wc);
+}
+
+static inline int __inline_iswblank(wint_t wc)
+{
+    return __inline_isblank(wc);
+}
+
+static inline int __inline_iswctype(wint_t, wctype_t)
+{
+    dbgln("FIXME: Implement iswctype()");
+    TODO();
+}
+
+static inline wctype_t __inline_wctype(const char*)
+{
+    dbgln("FIXME: Implement wctype()");
+    TODO();
+}
+
+static inline wint_t __inline_towlower(wint_t wc)
+{
+    return __inline_tolower(wc);
+}
+
+static inline wint_t __inline_towupper(wint_t wc)
+{
+    return __inline_toupper(wc);
+}
+
+static inline wint_t __inline_towctrans(wint_t, wctrans_t)
+{
+    dbgln("FIXME: Implement towctrans()");
+    TODO();
+}
+
+static inline wctrans_t __inline_wctrans(const char*)
+{
+    dbgln("FIXME: Implement wctrans()");
+    TODO();
+}
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+int iswalnum(wint_t wc);
+int iswalpha(wint_t wc);
+int iswcntrl(wint_t wc);
+int iswdigit(wint_t wc);
+int iswxdigit(wint_t wc);
+int iswspace(wint_t wc);
+int iswpunct(wint_t wc);
 int iswprint(wint_t wc);
+int iswgraph(wint_t wc);
+int iswlower(wint_t wc);
+int iswupper(wint_t wc);
+int iswblank(wint_t wc);
+int iswctype(wint_t wc, wctype_t desc);
+wctype_t wctype(const char* name);
+wint_t towlower(wint_t wc);
+wint_t towupper(wint_t wc);
+wint_t towctrans(wint_t wc, wctrans_t desc);
+wctrans_t wctrans(const char* str);
+
+#ifdef __cplusplus
+}
+#endif
+
+#define iswalnum __inline_iswalnum
+#define iswalpha __inline_iswalpha
+#define iswcntrl __inline_iswcntrl
+#define iswdigit __inline_iswdigit
+#define iswxdigit __inline_iswxdigit
+#define iswspace __inline_iswspace
+#define iswpunct __inline_iswpunct
+#define iswprint __inline_iswprint
+#define iswgraph __inline_iswgraph
+#define iswlower __inline_iswlower
+#define iswupper __inline_iswupper
+#define iswblank __inline_iswblank
+#define towlower __inline_towlower
+#define towupper __inline_towupper
+#define wctype __inline_wctype
+#define iswctype __inline_iswctype
+#define towctrans __inline_towctrans
+#define wctrans __inline_wctrans
```

Comment:
I'm not sure why this is required?
AFAIK these functions aren't supposed to be macros, and they're also normally not inline either

Reply:
Too be honest I just followed the same pattern as in [ctype.h](https://github.com/SerenityOS/serenity/blob/master/Userland/Libraries/LibC/ctype.h)

This commit from Gunnar adds macros for the ctype functions with reference as a fix for 'tr' port 03ffdeb4.

I should have added in my commit that I declared the wctype functions because of an issue with the 'seb' port getting confused with iswprint being declared twice. Seems like it expected more functions from wctype.h and then had a backup strategy of adding its own wctype.h. At least that is my theory since this branch now seemingly builds 'seb' fine

Reply:
I meant sed, not seb

Reply:
I think ctype _is_ normally done inline, since it's pretty commonly used.
I guess it's fine for now, though ideally we'd just stub them all out in one go...but that can wait I guess.

Reply:
So remove the macros, rename the inline functions in the header and define the function bodies in wctype.cpp?

I can do that

Reply:
yeah, let's do the less convoluted thing for now

Reply:
I think that should be it